src/bakabot/core/grades/grades.py

```diff
@@ -172,7 +172,17 @@ async def delete_grade_reaction(message: disnake.Message, emoji: disnake.message
                     pass
 
     @staticmethod
-    async def detect_changes(client: disnake.Client):
+    def handle_update_subjects_cache(grades: list[Grade], client: InteractionBot):
+        """Updates the SubjectsCache with the new subjects, if needed"""
+
+        subjects = [Subject(grade.subjectName, None) for grade in grades]
+
+        hasMadeChanges = SubjectsCache.handleUpdateSubjects(subjects)
+        if hasMadeChanges:
+            SubjectsCache.updateCommandsWithSubjects(client)
```

Comment:
Calls the `handleUpdateSubjects` for `SubjectCache` for subjects extracted from grades
- This is basically the same for subjects extracted from schedule